main.go

```diff
@@ -83,10 +83,13 @@ func main() {
 
 	if len(opts.Collectors) == 0 {
 		klog.Info("Using default collectors")
-		storeBuilder.WithEnabledResources(options.DefaultCollectors.AsSlice())
+		err = storeBuilder.WithEnabledResources(options.DefaultCollectors.AsSlice())
 	} else {
 		klog.Infof("Using collectors %s", opts.Collectors.String())
-		storeBuilder.WithEnabledResources(opts.Collectors.AsSlice())
+		err = storeBuilder.WithEnabledResources(opts.Collectors.AsSlice())
```

Comment:
Can you remove the code duplication in L86 and L89? We could assign a variable to hold either `options.DefaultCollectors` or `ops.Collectors` and then call `storeBuilder.WithEnabledResources` . once.